pkg/operation/botanist/controlplane/kube_scheduler.go

```diff
@@ -0,0 +1,472 @@
+	if k.secrets.Server.Name == "" || k.secrets.Server.Checksum == "" {
+		return fmt.Errorf("missing server secret information")
+	}
+
+	var (
+		configMap  = k.emptyConfigMap()
+		vpa        = k.emptyVPA()
+		service    = k.emptyService()
+		deployment = k.emptyDeployment()
+
+		labels = map[string]string{
+			v1beta1constants.LabelApp:  v1beta1constants.LabelKubernetes,
+			v1beta1constants.LabelRole: LabelScheduler,
+		}
+		labelsWithControlPlaneRole = utils.MergeStringMaps(labels, map[string]string{
+			v1beta1constants.DeprecatedGardenRole: v1beta1constants.GardenRoleControlPlane,
+		})
+		vpaUpdateMode = autoscalingv1beta2.UpdateModeAuto
+
+		port           = k.computeServerPort()
+		probeURIScheme = k.computeServerURIScheme()
+		env            = k.computeEnvironmentVariables()
+		command        = k.computeCommandLineFlags(port)
+	)
+
+	componentConfigYAML, componentConfigChecksum, err := k.computeComponentConfig()
+	if err != nil {
+		return err
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, k.client, configMap, func() error {
+		configMap.Data = map[string]string{KubeSchedulerDataKeyComponentConfig: componentConfigYAML}
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, k.client, vpa, func() error {
+		vpa.Spec.TargetRef = &autoscalingv1.CrossVersionObjectReference{
+			APIVersion: appsv1.SchemeGroupVersion.String(),
+			Kind:       "Deployment",
+			Name:       v1beta1constants.DeploymentNameKubeScheduler,
+		}
+		vpa.Spec.UpdatePolicy = &autoscalingv1beta2.PodUpdatePolicy{
+			UpdateMode: &vpaUpdateMode,
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, k.client, service, func() error {
+		service.Labels = labels
+		service.Spec.Selector = labels
+		service.Spec.Type = corev1.ServiceTypeClusterIP
+		service.Spec.Ports = kutil.ReconcileServicePorts(service.Spec.Ports, []corev1.ServicePort{
+			{
+				Name:     KubeSchedulerPortNameMetrics,
+				Protocol: corev1.ProtocolTCP,
+				Port:     port,
+			},
+		})
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, k.client, deployment, func() error {
+		deployment.Labels = labelsWithControlPlaneRole
+		deployment.Spec.Replicas = &k.replicas
+		deployment.Spec.RevisionHistoryLimit = pointer.Int32Ptr(0)
+		deployment.Spec.Selector = &metav1.LabelSelector{MatchLabels: labels}
+		deployment.Spec.Template = corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Annotations: map[string]string{
+					"checksum/configmap-componentconfig":           componentConfigChecksum,
+					"checksum/secret-" + k.secrets.Kubeconfig.Name: k.secrets.Kubeconfig.Checksum,
+					"checksum/secret-" + k.secrets.Server.Name:     k.secrets.Server.Checksum,
+				},
+				Labels: utils.MergeStringMaps(labelsWithControlPlaneRole, map[string]string{
+					v1beta1constants.LabelNetworkPolicyToDNS:            v1beta1constants.LabelNetworkPolicyAllowed,
```

Comment:
Do you also want to add v1beta1constants.LabelPodMaintenanceRestart: "true"? I already wanted to do it in another PR but then I saw this one 😄

Reply:
Sure, I can do it in a follow-up PR once this one is merged (to not mix a refactoring with a new feature) 👍

Reply:
Okay, thank you :)